### ISSUE TITLE
Problem with the intervals

### ISSUE THREAD

**Issue by pgillet** [HeidelTime/heideltime#35]
Hi, it's me again (you'll think I'm starting to annoy you),

I have a problem with the latest version 2.0.1 while tagging interval expressions in french.

For two "dumb" cases, I get the following results:

First example
------------------
Source = "Le jeudi 24 décembre"

```
<!DOCTYPE TimeML SYSTEM "TimeML.dtd">
<TimeML>
<TIMEX3INTERVAL earliestBegin="2015-12-24T00:00:00" latestBegin="2015-12-24T23:59:59" earliestEnd="2015-12-24T00:00:00" latestEnd="2015-12-24T23:59:59"><TIMEX3 tid="t2" type="DATE" value="2015-12-24">le jeudi 24 décembre</TIMEX3></TIMEX3INTERVAL>.
</TimeML>
```

2nd example
-----------------
Source = "Du lundi 21 décembre au vendredi 25 décembre"

```
<!DOCTYPE TimeML SYSTEM "TimeML.dtd">
<TimeML>
du <TIMEX3 tid="t5" type="DATE" value="2015-12-21">lundi 21 décembre</TIMEX3> au <TIMEX3 tid="t6" type="DATE" value="2015-12-25">vendredi 25 décembre</TIMEX3>.
</TimeML>
```
Regarding the first example, Heideltime found an interval for the first single date which spans from midnight to 12 PM.
This is correct but I'm expecting a single TIMEX3 with the DATE type, and this was the behavior before the project mavenization.

As for the 2nd example (from ... to ...), I obtain two dates while I'm expecting an interval (and I also used to get it before the mavenization).

My pom.xml:

```
...
        <!-- HeidelTime -->
        <dependency>
            <groupId>org.apache.uima</groupId>
            <artifactId>uimaj-core</artifactId>
            <version>2.8.1</version>
        </dependency>
        <dependency>
			<groupId>com.github.heideltime</groupId>
			<artifactId>heideltime</artifactId>
			<version>2.0.1</version>
		</dependency>
        <dependency>
            <groupId>edu.stanford.nlp</groupId>
            <artifactId>stanford-corenlp</artifactId>
            <version>3.3.1</version>
        </dependency>
...
```
I use Heideltime this way:

```
heidel = new HeidelTimeStandalone(Language.FRENCH, DocumentType.NEWS,
		OutputType.TIMEML, configPath, POSTagger.STANFORDPOSTAGGER,
		true);

...
```

Thanks again for your feedback,

Pascal

**Comment by jzell** [comment]
Hey Pascal,

unfortunately I can't look at this until Thursday. In the meantime, could you try to see if using an older version of UIMA fixes the issue? I don't think we changed anything about how the interval tagger works in the last 2-3 releases and the fact that this happened only after mavenization kinda points to dependencies in my head. 

Thanks, 
Julian

**Comment by pgillet** [comment]
Hi,

I tried several previous releases of UIMA, but I got the same result.
I also replaced the Stanford POSTagger by the TreeTagger, but it did not change anything as well.


'''
public class HeidelDemo {

    public static void main(String[] args)
	    throws DocumentCreationTimeMissingException {
	HeidelTimeStandalone heidel = new HeidelTimeStandalone(Language.FRENCH,
		DocumentType.NEWS, OutputType.TIMEML, "/tmp/config.props",
		POSTagger.TREETAGGER, true);

	Calendar cal = Calendar.getInstance();
	Date documentCreationTime = cal.getTime();

	String document = /* "Le jeudi 24 décembre"; */
	"Du lundi 21 décembre au vendredi 25 décembre";

	String result = heidel.process(document + ".", documentCreationTime);

	System.out.println(result);
    }

}
'''

**Comment by bogenpirat** [comment]
Hey Pascal,

thanks for your patience. I think I managed to fix this one. You can check out the snapshot i uploaded to nexus; first add the snapshots repo
```
     <repositories>
       <repository>
         <id>snapshots-repo</id>
         <url>https://oss.sonatype.org/content/repositories/snapshots</url>
         <releases><enabled>false</enabled></releases>
         <snapshots><enabled>true</enabled></snapshots>
       </repository>
     </repositories>
```
Then use the snapshot version
```
        <dependency>
            <groupId>com.github.heideltime</groupId>
            <artifactId>heideltime</artifactId>
            <version>2.0.1-SNAPSHOT</version>
        </dependency>
```
At least I think that's how you'd do that with Maven. Let me know if that helps.

**Comment by jzell** [comment]
Hey Pascal,

thanks for your patience. I think I managed to fix this one. You can check out the snapshot i uploaded to nexus; first add the snapshots repo
```
     <repositories>
       <repository>
         <id>snapshots-repo</id>
         <url>https://oss.sonatype.org/content/repositories/snapshots</url>
         <releases><enabled>false</enabled></releases>
         <snapshots><enabled>true</enabled></snapshots>
       </repository>
     </repositories>
```
Then use the snapshot version
```
        <dependency>
            <groupId>com.github.heideltime</groupId>
            <artifactId>heideltime</artifactId>
            <version>2.0.1-SNAPSHOT</version>
        </dependency>
```
At least I think that's how you'd do that with Maven. Let me know if that helps.

**Comment by pgillet** [comment]
Hi Julian,

Great, I am pleased to have identified a bug and to contribute to Heideltime, by giving you more work to do... ;)

I have the following remarks:
- The maven part is OK, and I managed to test the latest snapshot.
- I get an interval when one is expected (from date1 to date2), so I am fully satisfied with that.
- For single dates without time specified ("Thursday, December 24"), I still get an interval (in addition to a TIMEX3 DATE) which spans from midnight to 23:59 PM, but as said above, it is correct and I think it is more related to questions of design choices and philosophy of how Heideltime should behave in such cases.

But what about an expression such as "Le jeudi 24 décembre à 20h00" ("Thursday, December 24, at 8PM") with a time specified ?

The result is:
'''
<?xml version="1.0"?>
<!DOCTYPE TimeML SYSTEM "TimeML.dtd">
<TimeML>
<TIMEX3INTERVAL earliestBegin="2015-12-24T00:00:00" latestBegin="2015-12-24T23:59:59" earliestEnd="2015-12-24T00:00:00" latestEnd="2015-12-24T23:59:59"><TIMEX3 tid="t2" type="DATE" value="2015-12-24">Le jeudi 24 décembre</TIMEX3></TIMEX3INTERVAL> à <TIMEX3INTERVAL earliestBegin="2015-12-24T20:00:00" latestBegin="2015-12-24T20:00:59" earliestEnd="2015-12-24T20:00:00" latestEnd="2015-12-24T20:00:59"><TIMEX3 tid="t6" type="TIME" value="2015-12-24T20:00">20h00</TIMEX3></TIMEX3INTERVAL>
</TimeML>
'''

In this case, Heideltime should not be ambiguous and should only point to an instantaneous point on the timeline, which is 8PM on thursday, December 24.

Generally, Heideltime should differentiate:

- A date without time
- A time without date
- An instantaneous point on the time-line
- A full date and time
- Duration and Period: amounts of time
- Intervals: the time between two instants

But I do not know if it is compliant with the TimeML specification.

Thanks again,
Pascal

**Comment by jzell** [comment]
I'll need to look at some of the formatter code again to prevent single Timex3 annotations from getting the interval treatment, that used to work before and I neglected to test that aspect.

As for your example, that's a pretty clear case: There are two expressions extracted as per the french rules and they're extracted by an interval rule. That's perfectly normal behavior. Your example should be extracted differently in the first place though, so ideally we'd just create another time rule that catches the entire expression and normalizes all of the information. We don't have one that does that yet, and you can probably write a lot of rules before you've caught every time format that may come up.

In general, HeidelTime extracts and normalizes according to the TimeML 1.2 standard and iirc we try to keep with [these](http://www.timeml.org/tempeval2/tempeval2-trial/guidelines/timex3guidelines-072009.pdf) annotation practices. Under them, there are DATE/TIME/DURATION/SET annotations (no intervals) and we differentiate only between those four types and whether they (specifically DATE, TIME) are underspecified or explicit in terms of information. The interval logic was kind of tacked on and doesn't follow ANY standard, at least not in my limited recollection.

As noted above, I'll take a look at the singular interval annotations, but the other topic is probably best served if put into another issue report.

**Comment by pgillet** [comment]
OK, so I await your feedback for the single date expressions.

You should propose an extension to the TimeML spec to manage the intervals, and write an article about it :)

**Comment by JannikStroetgen** [comment]
According to TimeML, expressions such as this one "lundi 17 février, à 14 h 30" are annotated as two expressions:
<TIMEX3 tid="t7" type="DATE" value="2003-02-17">lundi 17 février</TIMEX3>, à <TIMEX3 tid="t8" type="TIME" value="2003-02-17T14:30">14 h 30</TIMEX3>
This is an example of the French TimeBank corpus. I kind of agree that one TIMEX for the full expression would be enough. However, typically prepositions do not belong inside the annotations....

We will think about it and let you know soon ;-)

**Comment by jzell** [comment]
@pgillet 5fe194f4b should fix the output

**Comment by pgillet** [comment]
Hi Julian,

There is no more jar in the snapshot repo.

Pascal

**Comment by jzell** [comment]
I have no idea what happened there. Try again, please, I see a release online now.

**Comment by pgillet** [comment]
That's perfect: I find the results that I was used to.

Thank you for your reactivity once again! :D

**Comment by jzell** [comment]
At long last. :-)
